torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -193,8 +195,6 @@ std::shared_ptr<FutureMessage> ProcessGroupAgent::sendImpl(
       futures_[requestId] = future;
     }
     message.setId(requestId);
-  } else {
-    future->markCompleted();
```

Comment:
From my understanding of the issue, we want to pass the future into `enqueueSend` and mark it with an exception message type if we catch an exception in the lambda. But the future API only lets us calls markCompleted once, so we'd have to remove this.